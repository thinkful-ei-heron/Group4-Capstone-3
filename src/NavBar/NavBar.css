.active {
    border-bottom: 5px rgb(110, 110, 110) inset;
    color: rgb(8, 184, 8) !important;
}

.active:hover {
    cursor: default;
}

.navbar-component {
    background-color: black !important;
    height: 2.3em;
    text-align: left;
}
.navbar-component a:hover {
    color: rgb(8, 184, 8);
}

.navbar-component input:checked ~ a {
    margin-bottom: 0;
}

.navbar-component {
    background-color: rgba(8, 8, 8, 0.774);
    height: 2.3em;
    text-align: left;
}
.navbar-component a, .navbar-component .active {
    display: block;
    position: relative;
    color: black;
    z-index: 4;
    top: 12px;
    left: 8px;
    font-weight: bold;
    font-family: 'Merriweather', serif;
    font-size: 15px !important;
    border-radius: 5px;
    padding: 8px;
    margin: 0;
    list-style-type: none;
    -webkit-font-smoothing: antialiased;
    transform-origin: 0 0;
    transform: translate(-150%, 0);
    transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0);
    width: 100px;
    background-color: rgba(226, 240, 227, 0.66);
    border: 1px solid rgba(60, 131, 64, 0.858);
    cursor: pointer;
    text-decoration: none;
}
.navbar-component input {
    display: inline-block;
    width: 60px;
    height: 50px;
    position: absolute;
    top: 190px;
    left: 4px;
    cursor: pointer;
    opacity: 0;
    z-index: 4;
    -webkit-touch-callout: none;
}

.navbar-component span {
    display: block;
    top: 6px;
    width: 33px;
    height: 4px;
    margin-bottom: 5px;
    position: relative;
    background: white;
    border-radius: 3px;
    z-index: 1;
    transform-origin: 4px 0;
    transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0),
    background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0), opacity 0.55s ease;
    left: 22px;
    margin-top: 4px;

}

.navbar-component span:first-child {
    transform-origin: 0 0;
}

.navbar-component a.active {
    border-bottom: none;
    text-shadow: 1px 1px 1px #000000;
}

.navbar-component input:checked ~ span {
    opacity: 1;
    transform: rotate(45deg) translate(-2px, -1px);
    background: white;
}



.navbar-component input:checked ~ .child-1 {
    transform: rotate(20deg) translate(0, -1px);
    position: relative;
    top: 15px;
}
.navbar-component input:checked ~ .child-2 {
    position: relative;
    top: 15px;
    transform-origin: 0 100%;
    transform: rotate(-20deg) translate(2px, -1px);
}
.navbar-component input:checked ~ .child-3 {
    opacity: 0;
    transform: rotate(0deg) scale(0.2, 0.2);
}


.navbar-component input:checked ~ a {
    transform: none;
}

@media (min-width: 480px) {
    .navbar-component {
        justify-content: space-evenly;
        display: inline-flex;
        width: 100vw;
    }
    .navbar-component input {
        display: none;
    }
    .navbar-component span {
        display: none;
    }
    .navbar-component a, .navbar-component .active {
        display: flex;
        position: relative;
        color: white;
        z-index: 4;
        top: unset;
        left: unset;
        font-weight: bold;
        font-family: 'Merriweather', serif;
        font-size: 1rem !important;
        border-radius: 0;
        padding: 8px;
        margin: auto 0;
        list-style-type: none;
        transform-origin: 0 0;
        transform: translate(0, 0);
        width: fit-content;
        background-color: unset;
        border: none;
        text-decoration: none;
    }
}


  

