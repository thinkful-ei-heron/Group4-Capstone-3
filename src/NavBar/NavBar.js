import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import UsersContext from '../../src/contexts/UserContext';
import './NavBar.css';

export default class NavBar extends Component {

  static contextType = UsersContext;

  handleLogoutClick = () => {
    this.context.processLogout()
  }

  render() {
    const links = [{to: '/home', name: 'Home'},{to: '/breweries', name: 'Breweries'},{to: '/add', name: 'Add Beer'}]
    return (
      <nav>
        {links.map((link, i) => <Link key={i} to={link.to} className={this.props.location.pathname === link.to ? 'active' : ''}>
          {link.name}
        </Link>)}
        <button onClick={this.handleLogoutClick}>
          Logout
        </button>
      </nav>
    )
  }
}