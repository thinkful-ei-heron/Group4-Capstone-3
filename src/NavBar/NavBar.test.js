import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import NavBar from './NavBar'

Enzyme.configure({ adapter: new Adapter() });

describe(`NavBar component`, () => {

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <NavBar location = {{pathname: ''}}/>
    </BrowserRouter>,
    div)
  ReactDOM.unmountComponentAtNode(div)
})
it('Displays NavBar when rendered', () => {
  const wrapper = shallow(<NavBar location = {{pathname: ''}} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})


})