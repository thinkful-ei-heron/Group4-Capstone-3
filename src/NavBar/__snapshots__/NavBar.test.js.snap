// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component Displays NavBar when rendered 1`] = `
<nav
  className="navbar-component"
>
  <input
    type="checkbox"
  />
  <span
    className="child-1"
  />
  <span
    className="child-2"
  />
  <span
    className="child-3"
  />
  <Link
    className=""
    key="0"
    to="/home"
  >
    Home
  </Link>
  <Link
    className=""
    key="1"
    to="/breweries"
  >
    Breweries
  </Link>
  <Link
    className=""
    key="2"
    to="/add"
  >
    Add Beer
  </Link>
  <Link
    onClick={[Function]}
    to="/"
  >
    Logout
  </Link>
</nav>
`;
